### PR TITLE
SED-2899 Housekeeping of time series and measurement data is slow with PostgreSQL

### PR DESCRIPTION
**Pull request by david-stephan**

### CODE REVIEW

**Comment by david-stephan** [comment]
@cl-exense I switched to LinkedHashSet to preserve the insertion order and meet the no duplicates requirements. As discussed the order being an integer comes from the fact that this is the format for mongodb. A refactoring should be done later  to use an enum instead.

File: step-framework-parent/step-framework-collections-postgresql/src/main/java/step/core/collections/postgresql/PostgreSQLCollection.java [hunk 1]
```diff
@@ -322,15 +329,24 @@ public void createOrUpdateCompoundIndex(String... fields) {
 
 	@Override
 	public void createOrUpdateCompoundIndex(Map<String, Integer> fields) {
+		LinkedHashSet<IndexField> indexFields = fields.entrySet().stream()
```
Comment (cl-exense) [inline step-framework-parent/step-framework-collections-postgresql/src/main/java/step/core/collections/postgresql/PostgreSQLCollection.java:332]: This is one of these cases where the resulting index field order is non-deterministic. Ideally, this should be deprecated and removed in the future, but for now it's ok according to the discussion.

File: step-framework-parent/step-framework-collections/src/main/java/step/core/collections/IndexField.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package step.core.collections;
+
+public class IndexField {
```
Comment (cl-exense) [inline step-framework-parent/step-framework-collections/src/main/java/step/core/collections/IndexField.java:3]: Does this have to be mutable? It looks a lot like a struct (or data-only) class that could be immutable, and thus be simplified to just have public final fields.

**Review by cl-exense — APPROVED** [review]
Approving, with minor nitpicks as always :-D

**Review by cl-exense — APPROVED** [review]

**Pull request merged**